Tools/SeeDot/seedot/Predictor/library_fixed.cpp

```diff
@@ -954,6 +954,64 @@ void Maxpool(MYINT *A, MYINT *B, MYINT N, MYINT H, MYINT W, MYINT C, MYINT FH, M
 	return;
 }
 
+// A = Normalise(A)
+void NormaliseL2(MYINT* A, MYINT N, MYINT H, MYINT W, MYINT C, MYINT scaleA, MYINT shrA) {
+	for (MYITE n = 0; n < N; n++) {
+		for (MYITE h = 0; h < H; h++) {
+			for (MYITE w = 0; w < W; w++) {
+		 
+
+				// calculate the sum square
+				int32_t sumSquare = 0;
+				MYINT shrAdiv = (1<<shrA);
+
+				for (MYITE c = 0; c < C; c++) {
+						MYINT tmp = (A[n * H * W * C + h * W * C + w * C + c] / shrAdiv);
+						sumSquare += tmp*tmp;
```

Comment:
this is a much less accurate way for integer division. will give problems in 8 bit code. 
can you try this: 

`int64_t tmp = (A[n * H * W * C + h * W * C + w * C + c]);
sumSquare += (MYINT)((tmp*tmp) / shrAdiv)`

or you can use something like matrix multiplication in this file (which uses the FASTAPPROX flag)

Reply:
done